packages/eas-cli/src/project/ios/target.ts

```diff
@@ -244,6 +256,43 @@ export function getApplePlatformFromSdkRoot(target: Target): ApplePlatform {
   } else if (sdkRoot.includes('macosx')) {
     return ApplePlatform.MAC_OS;
   } else {
+    return null;
+  }
+}
+
+/**
+ * Get Apple Platform from the Xcode TARGETED_DEVICE_FAMILY where possible.
+ *
+ * References:
+ * https://developer.apple.com/library/archive/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html
+ * https://stackoverflow.com/questions/39677524/xcode-8-how-to-change-targeted-device-family#comment100316573_39677659
+ *
+ * @returns - Apple Platform when known, defaults to null when unknown
+ */
+export function getApplePlatformFromDeviceFamily(target: Target): ApplePlatform | null {
+  const deviceFamily = target.buildSettings?.TARGETED_DEVICE_FAMILY;
+  if (!deviceFamily) {
+    return null;
+  }
+  if (typeof deviceFamily === 'string') {
```

Comment:
I would maybe suggest a simple lookup from a map... maybe make this change?

```patch
diff --git a/packages/eas-cli/src/project/ios/target.ts b/packages/eas-cli/src/project/ios/target.ts
index 0ccda087..0ee86e88 100644
--- a/packages/eas-cli/src/project/ios/target.ts
+++ b/packages/eas-cli/src/project/ios/target.ts
@@ -275,24 +275,18 @@ export function getApplePlatformFromDeviceFamily(target: Target): ApplePlatform
     return null;
   }
   if (typeof deviceFamily === 'string') {
-    const devices = deviceFamily.split(',');
-    const arbitraryDevice = devices[0];
-    const deviceFamilyNumber = Number(arbitraryDevice);
-    return deviceFamilyToPlatform(deviceFamilyNumber);
+    return deviceFamilyToPlatformMap[deviceFamily] || null;
   } else if (typeof deviceFamily === 'number') {
-    return deviceFamilyToPlatform(deviceFamily);
+    return deviceFamilyToPlatformMap[`${deviceFamily}`] || null;
   }
   throw new Error(
     `Unexpected device family type in XCode build settings: ${JSON.stringify(deviceFamily)}`
   );
 }
 
-function deviceFamilyToPlatform(deviceFamily: number): ApplePlatform | null {
-  if (deviceFamily === 1 || deviceFamily === 2) {
-    return ApplePlatform.IOS;
-  } else if (deviceFamily === 3) {
-    return ApplePlatform.TV_OS;
-  } else {
-    return null;
-  }
-}
+const deviceFamilyToPlatformMap: { [index: string]: ApplePlatform } = {
+  '1,2': ApplePlatform.IOS,
+  '1': ApplePlatform.IOS,
+  '2': ApplePlatform.IOS,
+  '3': ApplePlatform.TV_OS,
+};
```

Reply:
I don't think this lookup is a safe approach, this is a comma-separated list of values, so `2,1` is also valid

https://developer.apple.com/library/archive/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html

Reply:
Ah true... I've never seen `2,1` appear but it is indeed valid. Good catch :) I withdraw my suggestion